docs/content/users/extend/customization-extendibility.md

```diff
@@ -1,12 +1,69 @@
 ---
 search:
-  boost: 2
+  boost: 3
 ---
 
 # Extending and Customizing Environments
 
 DDEV provides several ways to customize and extend project environments.
 
+## Environment Variables for Containers and Services
+
+You can set custom environment variables in several places:
+
+1. An optional, project-level `.ddev/.env` file provides environment variables to all DDEV containers, including any additional services or add-ons. It can look something like this:
+
+    ```
+    MY_ENV_VAR='someval'
+    MY_OTHER_ENV_VAR='someotherval'
+    ```
+
+2. The global `web_environment` setting in `.ddev/global_config.yaml`.
+
+    ```yaml
+    web_environment:
+    - MY_ENV_VAR=someval
+    - MY_OTHER_ENV_VAR=someotherval
+    ```
+
+3. The project’s [`web_environment`](../configuration/config.md#web_environment) setting in `.ddev/config.yaml` or `.ddev/config.*.yaml`:
+
+    ```yaml
+    web_environment:
+    - MY_ENV_VAR=someval
+    - MY_OTHER_ENV_VAR=someotherval
+    ```
+
+If you’d rather use the CLI to set the project or global `web_environment` value, you can use the [`ddev config`](../usage/commands.md#config) command:
+
+```sh
+# Set MY_ENV_VAR for the project
+ddev config --web-environment-add="MY_ENV_VAR=someval"
+
+# Set MY_ENV_VAR globally
+ddev config global --web-environment-add="MY_ENV_VAR=someval
+```
+
+You can use the `--web-environment` flag to overwrite existing values rather than adding them.
+
+!!!warning "Don’t check in sensitive values!"
+Sensitive variables like API keys should not be checked in with your project. Typically you might use an `.env` file and _not_ check that in, but offer `.env.example` with expected keys that don’t have values. Some use global configuration for sensitive values, as that’s not normally checked in either.
```

Comment:
```suggestion
	Sensitive variables like API keys should not be checked in with your project. Typically you might use an `.env` file and _not_ check that in, but offer `.env.example` with expected keys that don’t have values. Some use global configuration for sensitive values, as that’s not normally checked in either.
```

Reply:
`.env.example` is ignored by `.ddev/.gitignore`.

We could mention how to add it to git here.

Reply:
Although this is something that people stumble on regularly, I'm always torn about how much we should teach about how many things.